packages/angular_devkit/build_angular/src/browser/index.ts

```diff
@@ -631,6 +629,24 @@ export function buildWebpackBrowser(
                   true,
                 );
               context.logger.info(bundleInfoText);
+
+              // Check for budget errors and display them to the user.
+              const budgets = options.budgets || [];
+              const budgetFailures = checkBudgets(budgets, webpackStats, processResults);
+              for (const {severity, message} of budgetFailures) {
+                switch (severity) {
+                  case ThresholdSeverity.Warning:
+                    webpackStats.warnings.push(message);
+                    break;
+                  case ThresholdSeverity.Error:
+                    webpackStats.errors.push(message);
```

Comment:
To ensure the compilation is considered a failure when an error occurs here, the following will need to be added after line 654 (in the error logger if block): `return { success: false };`

Reply:
Good catch. Updated to return failure.